backing: move the min votes threshold to the runtime

alindima opened the pull request:
Copy of https://github.com/paritytech/polkadot/pull/7577
Also addresses review comments.

For previous reviewers: I maintained the list of commits from previous PR. The only unreviewed commits, are the ones starting with: https://github.com/paritytech/polkadot-sdk/pull/1200/commits/e7e82bc11b50014f2886c87460b9225524e42d35, which address the review comments

eskimor commented on polkadot/runtime/parachains/src/configuration.rs:
```diff
@@ -292,6 +295,7 @@ impl<BlockNumber: Default + From<u32>> Default for HostConfiguration<BlockNumber
 			on_demand_fee_variability: Perbill::from_percent(3),
 			on_demand_target_queue_utilization: Perbill::from_percent(25),
 			on_demand_ttl: 5u32.into(),
+			minimum_backing_votes: 2,
```
Why not use the `LEGACY_MIN_BACKING_VOTES` constant here?

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3468303

alindima commented:
@sandreim @eskimor, could you please have another look? I addressed the review comments and I think that if it looks good, it could be merged.

Runtime upgrades are currently failing on Versi even with master code, so the reason is unrelated to this PR.

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3494434

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3494434

alindima commented:
> The CI pipeline was cancelled due to failure one of the required jobs. **Job name**: cargo-clippy **Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3494434

wut? the job passed..

eskimor commented on cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs:
```diff
@@ -588,6 +588,16 @@ impl RelayChainRpcClient {
 	}
 
+	/// Get the minimum number of backing votes for a candidate.
+	pub async fn parachain_host_minimum_backing_votes(
```
Is this even needed in cumulus? It should not.

alindima replied:
It doesn't compile without it, I see all runtime APIs have counterparts here.
I'm not sure why they're needed though

ordian replied:
I had similar question some time ago, and got this suggestion from @skunert: https://github.com/paritytech/cumulus/pull/2160#issuecomment-1562490741

alindima replied:
I think we can open a separate issue to discuss this further

alindima replied:
see: https://github.com/paritytech/polkadot-sdk/issues/1332

sandreim commented on polkadot/runtime/kusama/src/lib.rs:
```diff
@@ -1741,6 +1741,8 @@ pub mod migrations {
 
 		// Upgrade SessionKeys to include BEEFY key
 		UpgradeSessionKeys,
+
```
nit: a comment describing what the migration does is useful.
